src/DynamicDataTable.jsx

```diff
@@ -325,18 +325,38 @@ class DynamicDataTable extends Component {
     }
 
     renderLoadingTable() {
+        const { loadingIndicator, loadingMessage, loadingComponent } = this.props;
+
+        if (!!loadingComponent) {
+            return loadingComponent;
+        }
+
         return (
             <div className="table-responsive">
                 <table className="table table-striped">
                     <tbody>
                         <tr>
                             <td className="text-center">
-                                <div className="mt-5">
-                                    { this.props.loadingIndicator ? this.props.loadingIndicator : null }
-                                </div>
-                                <div className="mt-5">
-                                    { this.props.loadingMessage }
-                                </div>
+                                {!!loadingIndicator && !loadingMessage && (
+                                    <div>
+                                        { loadingIndicator }
+                                    </div>
+                                )}
+                                {!!loadingMessage && !loadingIndicator && (
+                                    <div>
+                                        { loadingMessage }
+                                    </div>
+                                )}
+                                {!!loadingMessage && !!loadingIndicator && (
+                                    <React.Fragment>
+                                        <div className="mt-5">
+                                            { loadingIndicator }
+                                        </div>
+                                        <div className="mt-5">
+                                            { loadingMessage }
+                                        </div>
+                                    </React.Fragment>
+                                )}
```

Comment:
Could this section just be done as follows?

```jsx
<React.Fragment>
    { loadingIndicator && (
        <div className="mt-5">
            { loadingIndicator }
        </div>
    ) }
    
    { loadingMessage && (
        <div className="mt-5">
            { loadingMessage }
        </div>
    ) }
</React.Fragment>
```

Reply:
This would also ensure there is a top margin if only one element is present.

Reply:
If the loading message looks fine without a top margin, feel free to remove it, and just do the following.

```jsx
<React.Fragment>
    { loadingIndicator ? (<div>{ loadingIndicator }</div>) : null }
    { loadingMessage ? (<div>{ loadingMessage }</div>) : null }
</React.Fragment>
```

Reply:
@DivineOmega but then you won't have margin between the message and indicator. How I have done it is to keep the original margin between the two (or non at all if only one is passed).

Reply:
@dextermb My thought is that if the user wants a margin between the two, they will be able to specify that in the `loadingIndicator` component instead?

Reply:
> @dextermb My thought is that if the user wants a margin between the two, they will be able to specify that in the `loadingIndicator` component instead?

Ah definitely a good suggestion! Let's do that then.